File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = 100001;
        int max = 0;
        for (int i = 0; i < prices.length; ++i) {
            min = Math.min(min, prices[i]);
            max = Math.max(max, prices[i] - min);
        }

        return max;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    '''
    7,1,5,3,6,4
    
    ans = 0



    '''
    def maxProfit(self, prices: List[int]) -> int:
        ans = 0
        l, r = 0, 1
        while r < len(prices):
            if prices[r] < prices[l]:
                l = r
            else:
                ans = max(ans, prices[r] - prices[l])
            r += 1
        return ans